Server/Ebenezer/User.cpp

```diff
@@ -8889,6 +8880,110 @@ bool CUser::GetStartPosition(int16_t* x, int16_t* z)
 	return false;
 }
 
+void CUser::SendClassPromotion(int16_t newClass)
+{
+	constexpr uint8_t classPromotionOpcode = 0x06;
+	char send_buff[32] = {};
+	int send_index = 0;
+
+	m_pUserData->m_sClass = newClass;
+	SetUserAbility();
+	Send2AI_UserUpdateInfo();
+	UserDataSaveToAgent();
+
+	SetByte(send_buff, WIZ_CLASS_CHANGE, send_index);
+	SetByte(send_buff, classPromotionOpcode, send_index);
+	SetShort(send_buff, m_pUserData->m_sClass, send_index);
+	SetShort(send_buff, _socketId, send_index);
+	Send(send_buff, send_index);
+	if (m_pMain != nullptr)
```

Comment:
We can assume `m_pMain` exists.
Additionally, since we're sending the same packet to the local player anyway, there's no reason to send it to us, then send it to everyone and exclude us.

Just send it to everyone -- including us -- instead.